src/main/java/org/truffleruby/core/rope/RopeOperations.java

```diff
@@ -213,13 +214,38 @@ public static LeafRope flatten(Rope rope) {
         return create(flattenBytes(rope), rope.getEncoding(), rope.getCodeRange());
     }
 
+    public static void visitNativeBytes(Rope rope, NativeBytesVisitor visitor) {
+        visitNativeBytes(rope, visitor, 0, rope.byteLength());
+    }
+
+    private final static ThreadLocal<Pointer> NATIVE_BYTES = ThreadLocal.withInitial(() -> Pointer.malloc(1024));
+
+    @TruffleBoundary
+    public static void visitNativeBytes(Rope rope, NativeBytesVisitor visitor, int offset, int length) {
+        // TODO CS 14-Nov-17 use a proper ropes visiting API
+
+        final byte[] bytes = flattenBytes(rope);
+
+        Pointer native_bytes = NATIVE_BYTES.get();
```

Comment:
Style: `native_bytes` should be camelCase

Reply:
Done.